frontend/gateway/grpcclient/client.go

```diff
@@ -354,6 +355,31 @@ func (r *reference) ReadFile(ctx context.Context, req client.ReadRequest) ([]byt
 	return resp.Data, nil
 }
 
+func (r *reference) ReadDir(ctx context.Context, req client.ReadDirRequest) ([]*fstypes.Stat, error) {
+	rdr := &pb.ReadDirRequest{
+		DirPath:        req.Path,
+		IncludePattern: req.IncludePattern,
+		Ref:            r.id,
+	}
+	resp, err := r.c.client.ReadDir(ctx, rdr)
```

Comment:
nit: would be good to check if client supports the cap and provide a nice error instead of letting it fail with grpc.

Reply:
I expected (but confess I didn't test) that the grpc error would be along the lines of `ReadDir not supported` I can make it say that the cap isn't supported though.